docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,202 @@
+Async Programming in C#/VB
+==========================
+By `Phillip Carter`_
+
+From a developer's perspective, the C#/VB async model is some nice syntax that hides complexity and allows you to focus on writing methods without worrying about things like thread pools and callbacks.
+
+The core of this are the ``async`` and ``await`` keywords (``Async`` and ``Await`` in VB.NET), which you sprinkle over otherwise normal methods. 
```

Comment:
I'm going to re-word this to state that it allows you to express your intent very easily and allows the code to remain simple.

To your point about candidates for async code, I think that should be an addition to the Overview document.  Although C#/VB and F# differ quite a bit in how async code is handled, they are fundamentally used for the same stuff (e.g. you wouldn't use async code to try and solve a CPU-bound task).